File: ch9/yojkim/ch9/franc_test.go
package ch9

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestFrancMultiplication(t *testing.T) {
	five := NewFranc(5)
	assert.Equal(t, NewFranc(10), five.Times(2))
	assert.Equal(t, NewFranc(15), five.Times(3))
}

func TestFrancEquality(t *testing.T) {
	assert.True(t, NewFranc(5).Equal(NewFranc(5)))
	assert.False(t, NewFranc(5).Equal(NewFranc(6)))
	assert.False(t, NewFranc(5).Equal(NewDollar(5)))
}

File: ch9/gunddol/src/ch9/Franc_Test.java
package ch9;

import org.junit.Test;

import static org.junit.Assert.*;

public class Franc_Test {

    @Test
    public void testFrancMultiplication() {
        Money five = Money.franc(5);
        assertEquals(Money.franc(10), five.times(2));
        assertEquals(Money.franc(15), five.times(3));
    }
}